src/lib/integration_test_cloud_engine/stack_driver_log_engine.ml

```diff
@@ -375,8 +380,72 @@ module Block_produced_query = struct
     ()
 end
 
+module Breadcrumb_added_query = struct
+  open Coda_base
+
+  module Result = struct
+    type t = {user_commands: User_command.t With_status.t list}
+  end
+
+  let filter testnet_log_filter =
+    String.concat ~sep:"\n"
+      [ testnet_log_filter
+      ; coda_container_filter
+      ; "\"Added breadcrumb user commands\"" ]
+
+  let parse json : Result.t Or_error.t =
+    let open Json_parsing in
+    let open Or_error.Let_syntax in
+    (* JSON path to metadata entry *)
+    let path = ["jsonPayload"; "metadata"; "user_commands"] in
+    let parser json =
```

Comment:
You can add this as a `Json_parsing.user_command` value so that we can reuse this later in other queries.